v1/client_test.go

```diff
@@ -50,6 +71,385 @@ func TestWriteSchemaCall(t *testing.T) {
+	require.Contains(resultBuffer, postTwo.ObjectId)
+	require.Len(resultBuffer, 2)
+}
+
+func TestLookupSubjects(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	client := testClient(t)
+	defer client.Conn.Close()
+
+	err := WriteTestSchema(client)
+	require.NoError(err)
+	emilia, beatrice, postOne, _, err := WriteTestTuples(client)
+	require.NoError(err)
+
+	// NOTE: we do a more naive approach here because the LookupSubjects API
+	// doesn't support cursoring.
+	resultBuffer := make([]string, 0)
+
+	response, err := client.PermissionsServiceClient.LookupSubjects(context.Background(), &v1.LookupSubjectsRequest{
+		SubjectObjectType: "user",
+		Permission:        "view",
+		Resource:          postOne,
+		Consistency:       fullyConsistent,
+	})
+	require.NoError(err)
+
+stream:
+	for {
+		item, err := response.Recv()
+		switch {
+		case errors.Is(err, io.EOF):
+			break stream
+		case err != nil:
+			require.NoError(err)
+		default:
+			resultBuffer = append(resultBuffer, item.Subject.SubjectObjectId)
+		}
+	}
+
+	require.Contains(resultBuffer, emilia.Object.ObjectId)
+	require.Contains(resultBuffer, beatrice.Object.ObjectId)
+	require.Len(resultBuffer, 2)
+}
+
+func TestCheckBulkPermissions(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	client := testClient(t)
+	defer client.Conn.Close()
+
+	err := WriteTestSchema(client)
+	require.NoError(err)
+	emilia, _, postOne, _, err := WriteTestTuples(client)
+	require.NoError(err)
+
+	response, err := client.PermissionsServiceClient.CheckBulkPermissions(context.Background(), &v1.CheckBulkPermissionsRequest{
+		Consistency: fullyConsistent,
+		Items: []*v1.CheckBulkPermissionsRequestItem{
+			{
+				Resource:   postOne,
+				Permission: "view",
+				Subject:    emilia,
+			},
+			{
+				Resource:   postOne,
+				Permission: "write",
+				Subject:    emilia,
+			},
+		},
+	})
+	require.NoError(err)
+
+	require.Len(response.Pairs, 2)
+	require.Equal(v1.CheckPermissionResponse_PERMISSIONSHIP_HAS_PERMISSION, response.Pairs[0].GetItem().Permissionship)
+	require.Equal(v1.CheckPermissionResponse_PERMISSIONSHIP_HAS_PERMISSION, response.Pairs[1].GetItem().Permissionship)
+}
+
+func TestBulkExportImport(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	client := testClient(t)
+	defer client.Conn.Close()
+
+	err := WriteTestSchema(client)
+	require.NoError(err)
+	_, _, _, _, err = WriteTestTuples(client)
+	require.NoError(err)
+
+	// Validate export
+	exportResponse, err := client.PermissionsServiceClient.ExportBulkRelationships(context.Background(), &v1.ExportBulkRelationshipsRequest{})
+	require.NoError(err)
+
+	exportResults := make([]*v1.Relationship, 0)
+stream:
+	for {
+		item, err := exportResponse.Recv()
+		switch {
+		case errors.Is(err, io.EOF):
+			break stream
+		case err != nil:
+			require.NoError(err)
+		default:
+			exportResults = append(exportResults, item.Relationships...)
+		}
+	}
+
+	require.Len(exportResults, 4)
```

Comment:
I don't see an obvious reason this would flake to 0 - that would mean the stream returned EOF immediately?

Reply:
Yeah, that's my understanding. Maybe I missed consistency somewhere?

Reply:
Yeah it's probably consistency. Lemme fix that.